### DISCUSSION TITLE
Add new file to target in a child project

### DISCUSSION THREAD

**Discussion by Almaz5200** [wojciech-kulik/xcodebuild.nvim#152] [Q&A] [answered]
I have a modularized project that uses Tuist to generate projects. Because of that, each feature is a project in itself. However, when I add a new file it only shows me the targets from the root projects. I can still add them using `tuist generate`, but it would be a lot quicker if it could detect those targets as well. Is there any way to do that?

**Comment by wojciech-kulik** [comment, marked as answer]
Have you tried to navigate in terminal to the folder with your subproject, open Neovim and configure xcodebuild.nvim specifically for this subproject? You could then switch between subprojects. If you use tmux or any other multiplexer, you could have a window for each project.

Currently, the plugin doesn't support multi-project apps :(

**Reply by Almaz5200** [reply]
That will work, but we've got a 100+ subprojects, so it's not really viable to switch between a 100 tmux panes :D. And at that point I might as well just run `tuist generate`.

Got you, thanks! Was hoping there was some kind of hack that would allow it to detect subprojects too :) For now I just will doing regenerations

**Reply by wojciech-kulik** [reply]
oh, right :D with this amount of projects this solution wouldn't make sense. I will think about it

**Reply by Almaz5200** [reply]
After some consideration I think it might be reasonable to, as a fallback in case regular "add to project" fails, to just start to search for .xcproject in a new file's directory and move up the file tree until you find one and add it to that project. It does feel doable, if I implemented that, would you accept a pull request with such changes?

**Reply by wojciech-kulik** [reply]
Sounds reasonable, but I think it would be best to stop navigating up once you hit your `cwd` to avoid going outside of your project's "bounds". And I think I would add it as an opt-in option rather than a default one.

**Reply by Almaz5200** [reply]
Agreed, we wouldn’t want to accidentally hijack something unrelated :D
Got you, I’ll make it as soon as I have time then

**Comment by Almaz5200** [comment]
Resolved by #188 :)
